### PR TITLE
Add ICD

### PR DESCRIPTION
**Pull request by GoodluckH**

### CODE REVIEW

File: docs/icd.md [hunk 1]
```diff
@@ -8,59 +8,88 @@
...
 --- | ---
-Status | Draft
+Status | :yellow_circle: Development
 
 ## Related Documents
 
 Document | Description
 --- | ---
-:construction: Requirements & User Stories :construction: | Requirements and user stories
+[Pricing Model](https://docs.google.com/spreadsheets/d/1mjPtaIn3E5m7r4nyKt_sJKG9BSFm2ty7Gzo7OqERxwo) | Unit economics and pricing mechanism of flights. The core logic of `svc-pricing` is largely derived from the pricing model.
```
Comment (ServiceDog) [inline docs/icd.md:26]: link the High-Level ICD when it's merged

File: docs/icd.md [hunk 2]
```diff
@@ -8,59 +8,88 @@
 
 ## Overview
 
-This document details the common software interfaces required by all services in the Arrow ecosystem.
+This document details the endpoints provided by the `svc-pricing` service.
+
+Currently, the server is configured to be live on `localhost:50051`, and is subject to change when deployed to production. This address can be interfaced by any API platforms like Postman, or even `curl`. 
+
+However, `svc-pricing` is intended to be consumed by other Arrow micro-services through the `svc-pricing-client` library, rather than the server being directly connected to and interacted with frontend clients.
```
Comment (ServiceDog) [inline docs/icd.md:15]: A bit confusing. You could state that autogenerated client-side functions for this gRPC server are conveniently packaged in the `svc-pricing-client` crate, to be used to ask for pricing by svc-cargo, svc-rideshare, and other customer-facing services.

**Review by ServiceDog — CHANGES_REQUESTED** [review]

File: docs/icd.md (hunk 1 above)
Comment (GoodluckH) [reply docs/icd.md:26]: [2baaed0](https://github.com/Arrow-air/svc-pricing/pull/2/commits/2baaed01064066accaa5ec859762f38787027c01)

GH actions to be re-run when that high-level ICD PR is merged.

File: docs/icd.md (hunk 2 above)
Comment (GoodluckH) [reply docs/icd.md]: [5559de2](https://github.com/Arrow-air/svc-pricing/pull/2/commits/5559de2b620e5c50c257f63d87648c96d6e7b3ca)

**Review by ServiceDog — APPROVED** [review]

**Review by ServiceDog — DISMISSED** [review]

File: .cargo/config.toml [hunk 3]
```diff
@@ -8,11 +8,9 @@
 rustflags = [
   "-Wmissing_docs",
   "-Wmissing_debug_implementations",
-  "-Wmissing_copy_implementations",
   "-Wtrivial_casts",
```
Comment (ServiceDog) [inline .cargo/config.toml:11]: Terraform will overwrite this next time it's updated, will break the pipeline. Would recommend restoring this file to default and addressing the warnings in build.rs and through allow statements

File: docs/icd.md [hunk 4]
```diff
@@ -8,59 +8,86 @@
...
 
 ## Related Documents
 
 Document | Description
 --- | ---
-:construction: Requirements & User Stories :construction: | Requirements and user stories
-
+[Pricing Model](https://docs.google.com/spreadsheets/d/1mjPtaIn3E5m7r4nyKt_sJKG9BSFm2ty7Gzo7OqERxwo) | Unit economics and pricing mechanism of flights. The core logic of `svc-pricing` is largely derived from the pricing model.
+[Uber Elevate White Paper](https://evtol.news/__media/PDFs/UberElevateWhitePaperOct2016.pdf) | Uber's research on UAM operations. Certain economic assumptions are referenced by Arrow's pricing model.
+[Services ICD](https://github.com/Arrow-air/se-services/blob/develop/docs/icd.md)
```
Comment (ServiceDog) [inline docs/icd.md:26]: Point to the svc-pricing conops and potentially the sdd

"Concept of Operations - `svc-pricing`" is how I'd label it

File: docs/icd.md [hunk 5]
```diff
@@ -8,59 +8,86 @@
...
 ## Frameworks
 
-See the Services ICD.
+See the [Services ICD](https://github.com/Arrow-air/se-services/blob/develop/docs/icd.md).
```
Comment (ServiceDog) [inline docs/icd.md:29]: Let's refer to it as the "High-Level Services ICD" to further distinguish this from the module's ICD

**Review by ServiceDog — CHANGES_REQUESTED** [review]

File: docs/icd.md (hunk 5 above)
Comment (GoodluckH) [reply docs/icd.md]: [7365da3](https://github.com/Arrow-air/svc-pricing/pull/2/commits/7365da3b4afe1903cd7f32c3dc28f7d1e1a1b0ac)

File: docs/icd.md (hunk 4 above)
Comment (GoodluckH) [reply docs/icd.md]: [7365da3](https://github.com/Arrow-air/svc-pricing/pull/2/commits/7365da3b4afe1903cd7f32c3dc28f7d1e1a1b0ac)

The link doesn't work for now until the #3 conops pr get merged.

File: .cargo/config.toml (hunk 3 above)
Comment (GoodluckH) [reply .cargo/config.toml]: [686a4cb](https://github.com/Arrow-air/svc-pricing/pull/2/commits/686a4cbfa3101ad1cb790b134423e597bc692766)

Very interesting. All I did was to add these flags back and the github actions are not complaining. NO idea if it's a false positive.

**Review by ServiceDog — APPROVED** [review]

**Pull request merged**